protokube/pkg/protokube/aws_volume.go

```diff
@@ -164,73 +164,83 @@ func newEc2Filter(name string, value string) *ec2.Filter {
 func (a *AWSVolumes) findVolumes(request *ec2.DescribeVolumesInput) ([]*Volume, error) {
```

Comment:
findVolumes is a wrapper around DescribeVolumes, which maps to our model.  But it doesn't do any filtering of mountable volumes (except in the error-case, i.e. we don't return mistagged volumes).  The idea is that we're doing the minimal mapping in the cloud abstraction layer.  The reason it matters is because of `AttachVolume`, which would now be broken for anything that isn't a mount of an etcd volume.  Admittedly that is more layering/design than practice, because the only volumes protokube mounts (today) are etcd volumes.

I'm thinking the right place is probably actually in the caller of FindVolumes, therefore, as then it addresses GCE as well :-)  (i.e. attachMasterVolumes)

But ... if you look at what's happening there, really the idea is that if you set the master tag on a volume, we're going to try mounting it on the masters.  I'm still not convinced that the behaviour we have isn't desireable ... if you add a master volume tag incorrectly, we hang; we could do a lot worse :-)